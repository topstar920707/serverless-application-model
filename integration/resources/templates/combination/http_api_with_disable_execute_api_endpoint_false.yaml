Resources:
  MyFunction:
    Type: AWS::Serverless::Function
    Properties:
      InlineCode: |
        exports.handler = async (event) => {
          const response = {
            statusCode: 200,
            body: JSON.stringify('Hello from Lambda!'),
          };
          return response;
        };
      Handler: index.handler
      Runtime: nodejs14.x
      Events:
        ImplicitGet:
          Type: HttpApi
          Properties:
            Method: Get
            Path: /get
            ApiId: 
              Ref: MyApi
        ImplicitPost:
          Type: HttpApi
          Properties:
            Method: Post
            Path: /post
            ApiId: 
              Ref: MyApi
  MyApi:
    Type: AWS::Serverless::HttpApi
    Properties:
      DisableExecuteApiEndpoint: False
      StageName: Prod
Outputs:
  ApiId:
    Value:
      Ref: MyApi  