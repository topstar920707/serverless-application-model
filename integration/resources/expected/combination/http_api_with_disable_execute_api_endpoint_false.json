[
  { "LogicalResourceId":"MyFunctionImplicitGetPermission", "ResourceType":"AWS::Lambda::Permission" },
  { "LogicalResourceId":"MyFunctionImplicitPostPermission", "ResourceType":"AWS::Lambda::Permission" },
  { "LogicalResourceId":"MyApi", "ResourceType":"AWS::ApiGatewayV2::Api" },
  { "LogicalResourceId":"MyApiProdStage", "ResourceType":"AWS::ApiGatewayV2::Stage" },
  { "LogicalResourceId":"MyFunction", "ResourceType":"AWS::Lambda::Function" },
  { "LogicalResourceId":"MyFunctionRole", "ResourceType":"AWS::IAM::Role" }
]